stats/thresholds.go

```diff
@@ -73,23 +53,67 @@ func newThreshold(src string, newThreshold *goja.Runtime, abortOnFail bool, grac
 		Source:           src,
 		AbortOnFail:      abortOnFail,
 		AbortGracePeriod: gracePeriod,
-		pgm:              pgm,
-		rt:               newThreshold,
+		Parsed:           parsedExpression,
 	}, nil
 }
 
-func (t Threshold) runNoTaint() (bool, error) {
-	v, err := t.rt.RunProgram(t.pgm)
-	if err != nil {
-		return false, err
+func (t *Threshold) runNoTaint(sinks map[string]float64) (bool, error) {
+	// Because aggregation method can either be a static keyword ("count", "rate", etc...),
+	// or a parametric expression ("p(somefloatingpointvalue)"), we need to handle this
+	// case specifically. If we encounter the percentile aggregation method token,
+	// we recompute the whole "p(value)" expression in order to look for it in the
+	// sinks.
+	sinkKey := t.Parsed.AggregationMethod
```

Comment:
What do you think about moving the logic (and the comment) of building the sink key to the `ThresholdExpression`, and then calling it `sinkKey := t.Parsed.SinkKey()` or `sinkKey := t.Parsed.Code()`.

The reason is that this calculation of the sink key is dependent exclusively on the state of `ThresholdExpression` and it also makes the current method a bit smaller. And that can be also re-used later in the:

```go
key := fmt.Sprintf("p(%v)", threshold.Parsed.AggregationValue)
```

Reply:
Great idea! On it 🦖